### ISSUE TITLE
Resumption tickets are version specific

### ISSUE THREAD

**Issue by huitema** [private-octopus/picoquic#1280]
See discussion in https://github.com/quicwg/quic-v2/pull/37

Also notice that servers should enforce that 0-RTT transport parameters used by client are the same as current server defaults.

**Comment by huitema** [comment]
In the current code, the ticket is composed as:
```
[64 bit random][encrypted-session-ticket][AEAD-tag]
```
The 64 bit number is used as ticket identifier, and as sequence number for AEAD encryption.

We have a requirement to carry more data for that ticket. 

Picoquic keeps a list of issued tickets, keyed by 64 bit ticket identifier. For each ticket, picoquic keeps EBMP information, used to resume the connection. In theory, we could add more data in the structure, e.g., indication of version. There are pros and cons to reusing this list, main con being that it is only used for performance, not security.

If we do not want to use the memorized ticket data, we need to add the version to the ticket. We are constrained by the decrypt API:
```
int (*call_back)(ptls_encrypt_ticket_t* encrypt_ticket_ctx,
    ptls_t* tls, int is_encrypt, ptls_buffer_t* dst, ptls_iovec_t src)
```
The Picotls code expect the decrypted ticket to be returned in the parameter `dst`:
```
typedef struct st_ptls_buffer_t {
    uint8_t *base;
    size_t capacity;
    size_t off;
    int is_allocated;
} ptls_buffer_t;
```
For decoding, the simplest solution is to allocate a big enough buffer, with a structure:
```
[64 bit random][encrypted-picoquic-ticket][encrypted-session-ticket][AEAD-tag]
```
After decrypting, picoquic can read and verify the parameters, then trim the offset.
For encrypting, the code will have to serialize:
```
[picoquic-ticket][session-ticket]
``
before encrypting the ticket.

If the picoquic ticket parameter are not as expected, then the decryption will fail.

**Comment by huitema** [comment]
Fixed in PR #1282
